python/tank/bootstrap/manager.py

```diff
@@ -221,11 +244,50 @@ def set_progress_callback(self, callback):
         """
         self._progress_cb = callback
 
-    def bootstrap_engine(self, engine_name, entity=None):
+    def bootstrap_engine(self,
```

Comment:
Back to good old `ToolkitManager.bootstrap_engine()` signature to execute the whole engine bootstrap logic synchronously in the main application thread.

Plus new `ToolkitManager.bootstrap_engine_async()` signature to bootstrap an Sgtk instance asynchronously in a background thread, followed by launching the engine synchronously in the main application thread.

Property `ToolkitManager.async_bootstrap` is gone.

When `ToolkitManager.bootstrap_engine_async()` cannot bootstrap asynchronously because Qt is not available, it falls back on a synchronous bootstrapping while still calling the provided callbacks in order for the client to work as expected.